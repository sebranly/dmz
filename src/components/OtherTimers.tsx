import * as React from 'react';
import { URL_DATA } from '../constants/general';
import { APITime, TimeType } from '../types';
import { Header } from './Header';
import { OpenClosedTimer } from './OpenClosedTimer';

export interface OtherTimersProps {
  className?: string;
  currentTimestamp: number;
}

const OtherTimers: React.FC<OtherTimersProps> = (props) => {
  const [times, setTimes] = React.useState<APITime[]>([]);

  const onMount = () => {
    const fetchAPITimes = async () => {
      try {
        const response = await fetch(URL_DATA);
        const data = await response.json();
        const safeTimes = (data.times || []) as APITime[];
        setTimes(safeTimes);
      } catch (error) {
        console.log(error);
      }
    };

    fetchAPITimes();
  };

  React.useEffect(() => {
    onMount();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const { className, currentTimestamp } = props;

  const timesBuilding21 = times.filter((time: APITime) => {
    const { name, type } = time;
    return type === TimeType.Map && name === 'Building 21';
  });

  return (
    <div className={className}>
      <Header text="Other Timers" />
      <div>
        <OpenClosedTimer currentTimestamp={currentTimestamp} times={timesBuilding21} />
      </div>
    </div>
  );
};

export { OtherTimers };
