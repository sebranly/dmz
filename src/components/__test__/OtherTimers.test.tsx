import * as React from 'react';
import { render } from '@testing-library/react';
import { OtherTimers } from '../OtherTimers';

test('OtherTimers', () => {
  const createComponent = () => <OtherTimers currentTimestamp={123} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
