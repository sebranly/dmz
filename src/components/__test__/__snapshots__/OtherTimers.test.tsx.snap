// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherTimers 1`] = `
NodeList [
  <div>
    <h3
      class="font-bold my-5 text-xl text-lime-500"
    >
      Other Timers
    </h3>
    <div />
  </div>,
]
`;
